wcfsetup/install/files/lib/system/Environment.class.php

```diff
@@ -0,0 +1,57 @@
+<?php
+
+namespace wcf\system;
+
+/**
+ * Provides functions to interact with the system/hosting enviroment.
+ *
+ * @author  Tim Duesterhus
+ * @copyright   2001-2021 WoltLab GmbH
+ * @license GNU Lesser General Public License <http://opensource.org/licenses/lgpl-license.php>
+ * @package WoltLabSuite\Core\System
+ */
+final class Environment
+{
+    public const SYSTEM_ID_REGISTRY_KEY = self::class . "::getSystemId()";
+
+    /**
+     * Returns a string representing key system information where
+     * changes might affect the software's behavior.
+     *
+     * Current information includes the location within the file system,
+     * the PHP minor version and the MySQL fork and minor version.
+     */
+    public static function getSystemId(): string
+    {
+        $ctx = \hash_init('sha256');
+
+        $sqlVersion = WCF::getDB()->getVersion();
+
+        foreach (
+            [
```

Comment:
I would move this into a separate variable in front of the `foreach()`, makes it easier to read.